client/file-transfer-service/src/handler.rs

```diff
@@ -524,18 +525,38 @@ impl FileTransferService {
                 };
 
                 if self.is_allowed(peer, file_key, bucket_id) {
+                    // Create a new channel for getting the result
+                    let (tx, mut rx) = tokio::sync::mpsc::channel::<bool>(1);
+
                     // Emit the event to the event bus, letting the upper layers know about the
                     // upload request.
                     self.emit(RemoteUploadRequest {
                         peer,
                         file_key,
                         file_key_proof,
                         bucket_id,
+                        file_complete_channel: tx,
                     });
 
+                    // Wait for result of whether the entire file is already stored by the provider.
+                    let file_complete = match rx.recv().await {
+                        Some(success) => success,
+                        None => {
+                            error!(
+                                target: LOG_TARGET,
+                                "Failed to receive upload request result from upper layers"
+                            );
+                            self.handle_bad_request(pending_response);
+                            return;
+                        }
+                    };
```

Comment:
@links234 wanted your specific input on the added channel field to the `RemoteUploadTask` task event.

Reply:
I didn't see another way to return a custom result type back from a specific event being processed by a task without introducing the channel to the event.

Reply:
this has been reverted